base/tests/test_OperationMultipleEvalInterModLinear.cpp

```diff
@@ -26,28 +27,28 @@ BOOST_AUTO_TEST_SUITE(TestOperationMultipleEvalInterModLinear)
 BOOST_AUTO_TEST_CASE(zeroAlpha) {
   srand(static_cast<unsigned>(time(nullptr)));
   const size_t dim = size_t(4+10.*rand()/static_cast<double>(RAND_MAX));
-  std::vector<std::vector<size_t>> interactions = std::vector<std::vector<size_t>>();
+  std::set<std::set<size_t>> interactions = std::set<std::set<size_t>>();
 
-  std::vector<size_t> tmp = std::vector<size_t> ();
+  std::set<size_t> tmp = std::set<size_t> ();
 
   // add empty interaction
-  interactions.push_back(tmp);
+  interactions.insert(tmp);
 
   // add all unit interactions
   for (size_t i = 0; i < dim; i++) {
-    tmp = std::vector<size_t> ();
-    tmp.push_back(i);
-    interactions.push_back(tmp);
+    tmp = std::set<size_t> ();
```

Comment:
Use `clear` to clear a set, similarly below.
```suggestion
    tmp.clear();
```

Reply:
There are still some left.